src/Illuminate/Support/Arr.php

```diff
@@ -290,7 +290,8 @@ public static function has($array, $key)
         }
 
         foreach (explode('.', $key) as $segment) {
-            if (! is_array($array) || ! array_key_exists($segment, $array)) {
+            if ((! is_array($array) || ! array_key_exists($segment, $array)) &&
+                (! $array instanceof ArrayAccess || ! $array->offsetExists($segment))) {
```

Comment:
Wouldn't this fail on any object that is not an ArrayAccess object and doesn't have an `offsetExists` method? Wouldn't you get an invalid method error?